src/node/interfaces.cpp

```diff
@@ -838,6 +839,49 @@ class ChainImpl : public Chain
     NodeContext& m_node;
 };
 
+class BlockTemplateImpl : public BlockTemplate
+{
+public:
+    explicit BlockTemplateImpl(std::unique_ptr<CBlockTemplate> block_template) : m_block_template(std::move(block_template)) {}
```

Comment:
In commit "Have createNewBlock return BlockTemplate interface" (6536b345a346670b8733e5372f988452727a2554)

Would be good to guard against undefined behavior by adding `assert(m_block_template);` in the constructor body and declaring `m_block_template` as `const unique_ptr` to prevent it being reset to null.

If this is done, the `Assert` in `getBlockHeader` could also be removed so it is more consistent with the other methods.

Reply:
Done